tests/unittests/config/test_cc_power_state_change.py

```diff
@@ -156,4 +163,29 @@ def check_lps_ret(psc_return, mode=None):
         raise Exception("\n".join(lines))
 
 
+class TestPowerStateChangeSchema:
+    @pytest.mark.parametrize(
+        "config, error_msg",
+        [
+            # Invalid mode
+            (
+                {"power_state": {"mode": "test"}},
+                r"'test' is not one of \['poweroff', 'reboot', 'halt'\]",
+            ),
+            # Delay can be a number, a +number, or "now"
+            ({"power_state": {"mode": "halt", "delay": "5"}}, None),
+            ({"power_state": {"mode": "halt", "delay": "now"}}, None),
+            ({"power_state": {"mode": "halt", "delay": "+5"}}, None),
+            ({"power_state": {"mode": "halt", "delay": "-5"}}, ""),
```

Comment:
This test should demonstrate why the pattern should use `{0,1}` instead of `*`.
```
           ({"power_state": {"mode": "halt", "delay": "++5"}}, ""),
```